pkg/build/builder/cmd/scmauth/passwd.go

```diff
@@ -0,0 +1,75 @@
+package scmauth
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	UsernamePasswordName = "passwd"
+	UserPassGitConfig    = `# username and password git config
+[url "%[1]s"]
+   insteadOf = %[2]s
+`
+)
+
+// SSHPrivateKey implements SCMAuth interface for using SSH private keys.
+type UsernamePassword struct {
+	SourceURL url.URL
+}
+
+// Setup creates a wrapper script for SSH command to be able to use the provided
+// SSH key while accessing private repository.
+func (u UsernamePassword) Setup(baseDir string) error {
```

Comment:
also allow pulling the username from the URL, e.g. "https://myuser@github.com/openshift/origin.git", and populating the password from the secret